scripts/constituents.py

```diff
@@ -484,124 +468,168 @@ def write_host_routines(cap, host, reg_funcname, init_funcname, num_const_funcna
+        cap.write(f"integer{spc} :: field_ind", 2)
+        cap.write(f"type({CONST_PROP_TYPE}), pointer :: const_prop => NULL()", 2)
+        # Declare dynamic constituent properties variables
+        for idx, scheme in enumerate(sorted(dyn_const_dict)):
+            cap.comment(f"dynamic constituent props variable for {scheme}", 2)
+            cap.write(f"type({CONST_PROP_TYPE}), allocatable :: dyn_const_prop_{idx}(:)", 2)
+        # end for
         cap.blank_line()
-        cap.write("{} = 0".format(herrcode), 2)
+        cap.write(f"{herrcode} = 0", 2)
         cap.write("num_consts = size(host_constituents, 1)", 2)
+        cap.write("num_dyn_consts = 0", 2)
         for suite in suite_list:
             const_dict = suite.constituent_dictionary()
             funcname = const_dict.num_consts_funcname()
             cap.comment(f"Number of suite constants for {suite.name}", 2)
             errvar_str = ConstituentVarDict.__errcode_callstr(herrcode,
                                                               herrmsg, suite)
             cap.write(f"num_suite_consts = {funcname}({errvar_str})", 2)
+            cap.write(f"if ({herrcode} /= 0) then", 2)
+            cap.write("return", 3)
+            cap.write("end if", 2)
             cap.write("num_consts = num_consts + num_suite_consts", 2)
         # end for
-        cap.write("if ({} == 0) then".format(herrcode), 2)
-        cap.comment("Initialize constituent data and field object", 3)
-        stmt = "call {}%initialize_table(num_consts)"
-        cap.write(stmt.format(const_obj_name), 3)
+        # Check for dynamic constituent routines
+        if len(dyn_const_dict) > 0:
+            cap.comment("Add in dynamic constituents", 2)
+            for idx, scheme in enumerate(sorted(dyn_const_dict)):
+                cap.write(f"call {dyn_const_dict[scheme]}(dyn_const_prop_{idx}, {obj_err_callstr})", 2)
+                cap.write(f"if ({herrcode} /= 0) then", 2)
+                cap.write("return", 3)
+                cap.write("end if", 2)
+                cap.write(f"num_dyn_consts = num_dyn_consts + size(dyn_const_prop_{idx})", 2)
+            # end for
+            cap.write("num_consts = num_consts + num_dyn_consts", 2)
+            cap.comment(f"Pack {dyn_const_name} array", 2)
+            cap.write(f"allocate({dyn_const_name}(num_dyn_consts), stat={herrcode})", 2)
+            cap.write(f"if ({herrcode} /= 0) then", 2)
+            cap.write(f"{herrmsg} = 'failed to allocate {dyn_const_name}'", 3)
+            cap.write("return", 3)
+            cap.write("end if", 2)
+            cap.write("index_start = 0", 2)
+            for idx, scheme in enumerate(sorted(dyn_const_dict)):
+                cap.write(f"do index = 1, size(dyn_const_prop_{idx}, 1)", 2)
+                cap.write(f"{dyn_const_name}(index + index_start) = dyn_const_prop_{idx}(index)", 3)
+                cap.write("end do", 2)
+                cap.write(f"index_start = size(dyn_const_prop_{idx}, 1)", 2)
```

Comment:
I see how this works if there are two entries in `dyn_const_dict` but what about 3? Shouldn't this be:
```suggestion
                cap.write(f"index_start = index_start + size(dyn_const_prop_{idx}, 1)", 2)
```

Reply:
yes good point! fixed